### ISSUE TITLE
configOverrides.extends breaks Node APIs

### ISSUE THREAD

**Issue by veej** [stylelint/stylelint#2280]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Trying to pass `configOverrides: { extends: 'stylelint-config-standard' }` to Node API doesn't seem to work (and it seems to break the linting process).

> What CSS is needed to reproduce this issue?

```css
.a {}
```

> What stylelint configuration is needed to reproduce this issue?

```
 {
  code: 'a { }',
  configBasedir: __dirname,
  config: {
    rules: {}
  },
  configOverrides: { extends: 'stylelint-config-standard' },
  syntax: 'scss',
  formatter: 'string'
}
```

> Which version of stylelint are you using?

`7.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Running the following script on Node:

```
import { lint } from 'stylelint';

lint({
  code: 'a { }',
  configBasedir: __dirname,
  config: {
    rules: {}
  },
  configOverrides: { extends: 'stylelint-config-standard' },
  syntax: 'scss',
  formatter: 'string'
})
  .then(({ output }) => console.log(output))
  .catch(err => console.log(err));
```

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

Stylelint should give something like the following:

<input css 1>
 1:3  ✖  Unexpected empty block   block-no-empty

> What actually happened (e.g. what warnings or errors you are getting)?

literally nothing is printed in console (not even errors)

**Comment by jeddy3** [comment]
@veej Thanks for using the issue template.

I think [you can have](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/node-api.md#configoverrides) _either_ a `config` property _or_ a `configOverrides` property. It doesn't make sense to have both together. 

> The difference between the `configOverrides` and `config` options is this: If any `config` object is passed, stylelint does not bother looking for a `.stylelintrc` file and instead just uses whatever `config` object you've passed; but if you want to *both* load a `.stylelintrc` file *and* override specific parts of it, `configOverrides` does just that.

If can you let us know exactly what outcome you're trying to achieve we can best advise you what options you should use?

If want to stylelint to search for a `.stylelintrc` file and then override it with a config, use:

```
{
  code: 'a { }',
  configOverrides: {
    rules: { 
      "block-no-empty": true
    }
  },
  syntax: 'scss',
  formatter: 'string'
}
```

If you don't want stylelint to search for a `.stylelintrc` file, then just use `config`:

```
{
  code: 'a { }',
  config: {
    rules: { 
      "block-no-empty": true
    }
  },
  syntax: 'scss',
  formatter: 'string'
}
```

Also, just add `extends: "stylelint-config-standard"` to your [configuration object](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/configuration.md#the-configuration-object) if you wish to extend it e.g.

```
{
  code: 'a { }',
  config: {
    extends: "stylelint-config-standard",
    rules: { 
      "block-no-empty": true
    }
  },
  syntax: 'scss',
  formatter: 'string'
}
```

> literally nothing is printed in console (not even errors)

I suspect that's because `config` is taking precedence over `configOverrides` and you've no rules defined for `config`. Use the `verbose` reporter to see what sources were checked e.g.

```
{
  code: 'a { }',
  config: {
    rules: {}
  },
  syntax: 'scss',
  formatter: 'verbose'
}
```

**Comment by jeddy3** [comment]
Umm, having said all that, you might have uncovered a bug.

```
{
  code: 'a { }',
  configOverrides: { 
    "extends": ["stylelint-config-standard"],
    "rules": {
      "block-no-empty": true,
    }
  },
  syntax: 'scss',
  formatter: 'verbose'
}
```

Displays nothing.

@davidtheclark Should you be able to extend from `configOverrides` config partial?

**Comment by veej** [comment]
Hey @jeddy3, thank you for your reply. 
As far as I can see from [tests](https://github.com/stylelint/stylelint/blob/master/lib/__tests__/standalone.test.js#L326), `configOverrides` **should** override config.
Moreover, as you uncovered in your previous post, even using a `.stylelintrc` instead of `config` property, `configOverrides` seems to break things.

What I'm trying to achieve is a sort of library aimed at sharing our basic configs and scripts (just like linting) between all our projects. This library will be included in all our projects and should provide, among other things, a script to lint our source files using a basic config, extensible through `.stylelintrc` files located in the root paths of our projects.
So, in my intentions, the script in the library would make use of `configOverrides` to provide the main config (through `extends`), allowing each project to override single rules via their `.stylelintrc` files.

**Comment by davidtheclark** [comment]
> Should you be able to extend from configOverrides config partial?

I do not think we've written any code to allow `extend` to work from `configOverrides`. This should be documented.

I don't think you *need* it, really, because when you are writing a Node module you should be able to write whatever code you need to put together the config object that you want. That idea makes me think we should deprecate that option: why have a `configOverrides` option when you could just use JavaScript to put things together according to your whims?

That said, if the only problem we have with `configOverrides` is `extends`, then I'd be open to a PR fixing that.

**Comment by veej** [comment]
@davidtheclark the issue also arise in [`stylelint-webpack-plugin`](https://github.com/JaKXz/stylelint-webpack-plugin) that, as far I can see, uses the Node APIs.
In webpack config is a little bit more overwhelming to implement the config merging logic.
